mongo_orchestration/servers.py

```diff
@@ -308,8 +309,23 @@ def start(self, timeout=300):
                     "Server did not respond to 'isMaster' after %d attempts."
                     % timeout)
         except (OSError, TimeoutError):
-            logger.exception("Could not start Server.")
-            raise
+            logpath = self.cfg.get('logpath')
+            if logpath:
+                # Copy the server logs into the mongo-orchestration logs.
+                logger.error(
+                    "Could not start Server. Please find server log below.\n"
+                    "=====================================================")
+                with open(logpath) as lp:
+                    logger.error(lp.read())
```

Comment:
The log file is new for each run, at least in jenkins.

This also only happens if the mongod/s process fails to start for some
reason, which probably indicates that the log file is small. "logappend"
isn't enabled by default by Mongo Orchestration, at least.

On Thu, Sep 24, 2015 at 5:19 PM, Bernie Hackett <notifications@github.com>
wrote:

> In mongo_orchestration/servers.py
> <https://github.com/10gen/mongo-orchestration/pull/199#discussion_r40389203>
> :
>
> > @@ -308,8 +309,23 @@ def start(self, timeout=300):
> >                      "Server did not respond to 'isMaster' after %d attempts."
> >                      % timeout)
> >          except (OSError, TimeoutError):
> > -            logger.exception("Could not start Server.")
> > -            raise
> > +            logpath = self.cfg.get('logpath')
> > +            if logpath:
> > +                # Copy the server logs into the mongo-orchestration logs.
> > +                logger.error(
> > +                    "Could not start Server. Please find server log below.\n"
> > +                    "=====================================================")
> > +                with open(logpath) as lp:
> > +                    logger.error(lp.read())
>
> I'm a little worried that this could be a huge amount of data if the
> logfile isn't new for each run. Should I worry?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/10gen/mongo-orchestration/pull/199/files#r40389203>.
>